native-engine/datafusion-ext-plans/src/agg/agg_buf.rs

```diff
@@ -633,6 +700,36 @@ impl AggDynValue for AggDynSet {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub enum OptimizeSet {
+    Empty,
+    LitteVec(SmallVec<[ScalarValue; 4]>),
+    Set(HashSet<ScalarValue>),
+}
+
+impl Default for OptimizeSet {
+    fn default() -> Self {
+        OptimizeSet::Empty
+    }
+}
+
+impl OptimizeSet {
+    fn mem_size(&self) -> usize {
+        match self {
+            OptimizeSet::LitteVec(vec) => {
+                std::mem::size_of_val(vec)
```

Comment:
can be simplified to `std::mem::size_of_val(vec) + vec.iter().map(|sv| sv.size()).sum::<usize>()`